app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -1545,5 +1550,55 @@ public void onNewMessages(Message[] messages) {
         if (narrowedList != null) {
             narrowedList.onNewMessages(messages);
         }
+        showSnackbarNotification(messages); //Show notification
+    }
+
+    Snackbar snackbar;
+
+    private void setupSnackBar() {
+        final CoordinatorLayout coordinatorLayout = (CoordinatorLayout) findViewById(R.id.coordinatorLayout);
+        snackbar = Snackbar.make(coordinatorLayout, "", Snackbar.LENGTH_LONG);
+        View v = snackbar.getView();
+        CoordinatorLayout.LayoutParams params = (CoordinatorLayout.LayoutParams) v.getLayoutParams();
+        params.gravity = Gravity.TOP;
+        v.setLayoutParams(params);
+    }
+
+    NarrowFilter narrowFilter;
+
+    private void showSnackbarNotification(Message[] messages) {
+        String prevId = messages[0].getIdForHolder();
+        for (int i = 1; i < messages.length; i++) {
+            if (!prevId.equals(messages[0].getIdForHolder())) {
+                prevId = null;
+                break;
+            }
+        }
+        if (prevId == null) {
+            snackbar.setText(String.format(Locale.getDefault(), getString(R.string.new_message_mul_sender), messages.length, (messages.length > 1 ? "s" : "")));
+            narrowFilter = null;
+            if (narrowedList != null) {
+                narrowedList = null;
+                getSupportFragmentManager().popBackStack(NARROW,
+                        FragmentManager.POP_BACK_STACK_INCLUSIVE);
+            }
+            snackbar.setAction(R.string.SHOW, new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    homeList.showLatestMessages();
+                }
+            });
+        } else {
+            String name = (messages[0].getType() == MessageType.PRIVATE_MESSAGE) ? String.format(Locale.getDefault(), getString(R.string.notify_private), messages[0].getSenderFullName()) : String.format(Locale.getDefault(), getString(R.string.notify_stream), messages[0].getStream().getName() , messages[0].getSubject());
```

Comment:
the built-in function `getString()` will handle string formatting for you.  this will cut down on the length of this line a lot.

in other words you can re-write this:
```
String.format(Locale.getDefault(), getString(R.string.notify_private), messages[0].getSenderFullName())
```

as this
```
getString(R.string.notify_private, messages[0].getSenderFullName())
```

Docs: https://developer.android.com/reference/android/content/Context.html#getString(int, java.lang.Object...)